### ISSUE TITLE
AWS App Mesh now supports HTTP2 and gRPC services

### ISSUE THREAD

**Issue by ewbankkit** [terraform-providers/terraform-provider-aws#10803]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS App Mesh now supports HTTP2 and gRPC services.

### New or Affected Resource(s)

* [`aws_appmesh_route`](https://www.terraform.io/docs/providers/aws/r/appmesh_route.html)

### Potential Terraform Configuration

```hcl
resource "aws_appmesh_route" "serviceb" {
  grpc_route {
    action {
    }

    match {
    }
  }
}
```

### References

[Announcement](https://aws.amazon.com/about-aws/whats-new/2019/11/aws-app-mesh-now-supports-http2-and-grpc-services/).
[User guide](https://docs.aws.amazon.com/app-mesh/latest/userguide/routes.html).
[API reference](https://docs.aws.amazon.com/app-mesh/latest/APIReference/API_GrpcRoute.html)

Requires [AWS SDK v1.25.19](https://github.com/aws/aws-sdk-go/releases/tag/v1.25.19):
* https://github.com/terraform-providers/terraform-provider-aws/pull/10476

AWS App Mesh roadmap:
* https://github.com/aws/aws-app-mesh-roadmap/issues/96

Related:
* https://github.com/terraform-providers/terraform-provider-aws/issues/9802
* https://github.com/terraform-providers/terraform-provider-aws/issues/10075

**Comment by dominhhai** [comment]
Is any plan to support this?

**Comment by tilyupo** [comment]
[aws_appmesh_virtual_node](https://www.terraform.io/docs/providers/aws/r/appmesh_virtual_node.html#protocol) affected as well. Listeners now support grpc and http2

**Comment by hatstand** [comment]
[`aws_appmesh_virtual_node`](https://www.terraform.io/docs/providers/aws/r/appmesh_virtual_node.html) is also affected by this as `.spec.listener.port_mapping[].protocol` can contain `http2` or `grpc` too now.

**Comment by ewbankkit** [comment]
@hatstand Thanks for noticing that.
I have updated https://github.com/terraform-providers/terraform-provider-aws/pull/11669 to add gRPC and HTTP/2 support to `aws_appmesh_virtual_node` listeners.

**Comment by hatstand** [comment]
Thanks for adding that to the PR @ewbankkit

**Comment by lays147** [comment]
Hi, `aws_appmesh_virtual_router` should also support the grpc and http/2 protocols. As I mentioned on issue #15198 =D

**Comment by ewbankkit** [comment]
@lays147 Yes, you are correct. The changes to `aws_appmesh_virtual_router` have been implemented in the PR that closes this issue, I will change the title of that PR (and update this issue to capture the additional resource).
Thanks.
